backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/codec/impl/LuceneStringFieldCodec.java

```diff
@@ -49,7 +52,14 @@ public void encode(LuceneDocumentBuilder documentBuilder, String absoluteFieldPa
 			return;
 		}
 
-		documentBuilder.addField( new Field( absoluteFieldPath, value, fieldType ) );
+		if ( searchable ) {
+			// searchable either projectable or not projectable
+			documentBuilder.addField( new Field( absoluteFieldPath, value, fieldType ) );
+		}
+		else if ( fieldType.stored() ) {
+			// not searchable and projectable
+			documentBuilder.addField( new StoredField( absoluteFieldPath, normalize( absoluteFieldPath, value ) ) );
+		}
```

Comment:
Not sure this is necessary? The `fieldType` should already take that into account.

See `org.hibernate.search.backend.lucene.types.dsl.impl.LuceneStringIndexFieldTypeContext#getFieldType`.

In particular, if the field is not searchable, `indexOptions` should be forced to `IndexOptions.NONE`.
You could disable norms and term vectors too, at least if they are not explicitly set, but I'm not sure it matters if `indexOptions` is set to `NONE`.

Reply:
Ok. It makes sense to handle both projectable and searchable using the type created by `LuceneStringIndexFieldTypeContext#getFieldType`. Let's do that.